src/Platform/Microsoft.Testing.Platform/TestHost/TestHostManager.cs

```diff
@@ -57,35 +58,37 @@ internal async Task<ActionResult<ITestFrameworkInvoker>> TryBuildTestAdapterInvo
         return ActionResult.Fail<ITestFrameworkInvoker>();
     }
 
-    public void AddTestExecutionFilterFactory(Func<IServiceProvider, ITestExecutionFilterFactory> testExecutionFilterFactory)
+    public async Task<ITestExecutionFilter> BuildFilterAsync(
+        IServiceProvider serviceProvider,
+        ICollection<TestNode>? testNodes)
     {
-        Guard.NotNull(testExecutionFilterFactory);
-        if (_testExecutionFilterFactory is not null)
+        if (testNodes?.Count > 0)
         {
-            throw new InvalidOperationException(PlatformResources.TEstExecutionFilterFactoryFactoryAlreadySetErrorMessage);
+            return new TestNodeUidListFilter(testNodes.Select(x => x.Uid).ToArray());
```

Comment:
Is the intent here to run whatever was selected in IDE, even if there is some other filter that would filter it out?

Reply:
Yeah exactly. `TestNodeUidListFilter` seems to be special and work a bit differently to other filters.

Alternatively I could add it to the built filters list and combine it with others in an aggregate filter (if any)?

But not sure that's even supported at the moment if its only IDEs/Test Explorer driving this.

But if it is going to be extended to the CLI in the future then it might make sense.

Reply:
I tend to think `TestHostManager` could have two `BuildFilterAsync` overloads. One that only takes `ICollection<TestNode>? testNodes` which is only used by `ServerTestHost`, and the signature kinda implies that it only considers the given test nodes, and another overload that only takes `IServiceProvider serviceProvider` which is used by ConsoleTestHost.

So in future, if we decided to add `TestNodeUidListFilter` for CLI scenarios, we go through the proper code path that combines all filters properly.

Reply:
Added a separate method for the test node scenario